src/main/java/net/dries007/tfc/common/blocks/devices/PowderkegBlock.java

```diff
@@ -0,0 +1,172 @@
+
+    private static final VoxelShape SHAPE = box(2, 0, 2, 14, 16, 14);
+
+    public static void toggleSeal(Level level, BlockPos pos, BlockState state)
+    {
+        level.getBlockEntity(pos, TFCBlockEntities.POWDERKEG.get()).ifPresent(powderkeg -> {
+            final boolean previousSealed = state.getValue(SEALED);
+            level.setBlockAndUpdate(pos, state.setValue(SEALED, !previousSealed));
+            if (previousSealed)
+            {
+                powderkeg.onUnseal();
+            }
+            else
+            {
+                powderkeg.onSeal();
+            }
+        });
+    }
+
+    public PowderkegBlock(ExtendedProperties properties)
+    {
+        super(properties, DeviceBlock.InventoryRemoveBehavior.SAVE);
+        registerDefaultState(getStateDefinition().any().setValue(SEALED, false));
+    }
+
+    @Override
+    @SuppressWarnings("deprecation")
+    public VoxelShape getShape(BlockState state, BlockGetter level, BlockPos pos, CollisionContext context)
+    {
+        return SHAPE;
+    }
+
+    @Override
+    @SuppressWarnings("deprecation")
+    public InteractionResult use(BlockState state, Level level, BlockPos pos, Player player, InteractionHand hand, BlockHitResult hit)
+    {
+        final PowderkegBlockEntity powderkeg = level.getBlockEntity(pos, TFCBlockEntities.POWDERKEG.get()).orElse(null);
+        if (powderkeg != null)
+        {
+            final ItemStack stack = player.getItemInHand(hand);
+            if (stack.isEmpty() && player.isShiftKeyDown())
+            {
+                toggleSeal(level, pos, state);
+                level.playSound(null, pos, SoundEvents.WOOD_PLACE, SoundSource.BLOCKS, 1.0f, 0.85f);
+                return InteractionResult.SUCCESS;
+            }
+            else if (!level.isClientSide() && player instanceof ServerPlayer serverPlayer)
+            {
+                NetworkHooks.openGui(serverPlayer, powderkeg, powderkeg.getBlockPos());
+            }
+            return InteractionResult.SUCCESS;
+        }
+        return InteractionResult.PASS;
+    }
+
+    @Override
+    public BlockState getStateForPlacement(BlockPlaceContext context)
+    {
+        return context.getItemInHand().getTag() != null ? defaultBlockState().setValue(SEALED, false) : defaultBlockState();
+    }
+
+    @Override
+    public void appendHoverText(ItemStack stack, @Nullable BlockGetter level, List<Component> tooltip, TooltipFlag flag)
+    {
+        final CompoundTag tag = stack.getTagElement(Helpers.BLOCK_ENTITY_TAG);
+        if (tag != null)
+        {
+            final CompoundTag inventoryTag = tag.getCompound("inventory");
+            final ItemStackHandler inventory = new ItemStackHandler();
+
+            inventory.deserializeNBT(inventoryTag.getCompound("inventory"));
+
+            tooltip.add(new TranslatableComponent("tfc.tooltip.contents").withStyle(ChatFormatting.DARK_GREEN));
+            Helpers.addInventoryTooltipInfo(inventory, tooltip);
```

Comment:
Wrap this in a `if (Helpers.isEmpty(inventory)`, so the "Contents" label isn't shown if the keg is empty.